pkg/apis/greenhouse/v1alpha1/plugindefinition_types.go

```diff
@@ -111,6 +111,15 @@ func (p *PluginOption) IsValid() error {
 	case PluginOptionTypeMap:
 		var m map[string]any
 		return json.Unmarshal(p.Default.Raw, &m)
+	case PluginOptionTypeSecret:
+		var s string
+		if err := json.Unmarshal(p.Default.Raw, &s); err != nil {
+			return err
+		}
+		if s != "" {
```

Comment:
It is sufficient to check for the length of `Default.Raw` to make sure the value was not set
```suggestion
		if len(p.Default.Raw) != 0 {
```

Reply:
:heavy_check_mark:

Reply:
Hmm tests fail after this merge.

The `Raw` representation for 
- `nil` is `[]uint8 len: 4, cap: 8, [110,117,108,108]` (a.k.a. `"null"`)   
- `""` is  `[]uint8 len: 2, cap: 8, [34,34]`  (a.k.a. `"\"\""`)

Both `len(p.Default.Raw)>0` if we do not unmarshal the json first. I think this was the reason why I added the unmarshalling in the first place...

I added a second (also failing) test to illustrate. 
We could stop using `json.Marshal(val)` for `val==nil` in the tests, which is done by the `test.MustReturnJSONFor()` func.
But I would prefer to catch the wider range of _not defaulted_ string values?

Reply:
My Bad, I had just tested this with by converting `""` to `[]byte` and that looked good. Yeah let's do it the way you have proposed intiially, then we are on the safe side.